packages/uicore/src/components/Filters/components/FilterSingleSelectDropDown/FiltersSelectDropDown.tsx

```diff
@@ -153,19 +159,46 @@ export function FiltersSelectDropDown(props: FilterSelectDropDownProps): React.R
 
   const renderMenu: ItemListRenderer<SelectOption> = ({ items: itemsToRender, itemsParentRef, renderItem }) => {
     let renderedItems
+
     if (loading || isLoading) {
       renderedItems = (
         <li className={css.menuItem} style={{ justifyContent: 'center' }}>
           <Spinner size={20} />
         </li>
       )
     } else if (itemsToRender.length > 0) {
-      renderedItems = filterItems(itemsToRender)
-        .map(renderItem)
-        .filter(item => item !== null)
+      const filteredItems = filterItems(itemsToRender)
+
+      if (sections) {
+        renderedItems = Object.keys(sections).map((sectionTitle, index) => {
+          const sectionItems = sections[sectionTitle]
+          const sectionFilteredItems = filteredItems.filter(item => sectionItems.includes(item.value as string))
+
+          return sectionFilteredItems.length > 0 ? (
+            <React.Fragment key={index}>
+              <MenuDivider title={sectionTitle} className={css.sectionTitle} />
+              {sectionFilteredItems.map(renderItem)}
+            </React.Fragment>
+          ) : null
+        })
+
+        // Render items not in any section
+        const uncategorizedItems = filteredItems.filter(
+          item => !Object.values(sections).some(sectionItems => sectionItems.includes(item.value as string))
+        )
```

Comment:
In this logic, `Object.values(sections)` is recalculated in every iteration of the `filter`. You can move out of this loop to avoid redundant calculations 👇🏻 


```js
const allSectionItems = Object.values(sections).flat()
const uncategorizedItems = filteredItems.filter(
  item => !allSectionItems.includes(item.value as string)
)
```